### PR TITLE
LCAM-1581|Response mapping refactoring

### PR DESCRIPTION
**Pull request by venkat980**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LCAM-1581)

Refactored the ApiMaatCalculateContributionResponse mapping to avoid duplication and centralise the response mapping code in one place.

Renamed and refactored the methods in MaatCalculateContributionService to be more meaningful.

Update the unit tests as per the refactored code.

Renamed the applId in Contribution class as applicantId to be in-line with changes in other repos.

## Checklist

Before you ask people to review this PR:

- [ ] Tests should be passing: `./gradlew test`
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [ ] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [ ] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

## Additional checks

- Don’t forget to [run](https://github.com/ministryofjustice/laa-crimeapps-maat-functional-tests/actions/workflows/ExecuteUiTests.yaml) the MAAT functional test suite after deploying your changes to the DEV or TEST environments to ensure your changes haven’t broken any of the functional tests.

### CODE REVIEW

**Review by vijay-laa — APPROVED** [review]

**Review by karlbaker02 — COMMENTED** [review]
Looks good, only a few minor comments.

File: crown-court-contribution/src/main/java/uk/gov/justice/laa/crime/contribution/service/MaatCalculateContributionService.java [hunk 1]
```diff
@@ -206,75 +187,67 @@ public ApiMaatCalculateContributionResponse getCalculateContributionResponse(
...
     }
 
-    public ApiMaatCalculateContributionResponse doContribs(final CalculateContributionDTO calculateContributionDTO,
-                                                           final ContributionResponseDTO contributionResponseDTO,
-                                                           final String fullResult,
-                                                           final RepOrderDTO repOrderDTO) {
-        ApiMaatCalculateContributionResponse response;
-        log.info("doContribs");
+    private static String getCrownCourtOutcome(CalculateContributionDTO calculateContributionDTO) {
+        if (null != calculateContributionDTO.getCrownCourtOutcomeList()
```
[the remark this answers is not in the document]
Comment (venkat980) [reply crown-court-contribution/src/main/java/uk/gov/justice/laa/crime/contribution/service/MaatCalculateContributionService.java]: I have refactored this before your review, can you please take a look at the latest commit. Thanks!

File: crown-court-contribution/src/main/java/uk/gov/justice/laa/crime/contribution/service/MaatCalculateContributionService.java [hunk 2]
```diff
@@ -143,19 +143,15 @@ public static LocalDate getEffectiveDate(final CalculateContributionDTO calculat
...
 
         RepOrderDTO repOrderDTO = maatCourtDataService.getRepOrderByRepId(calculateContributionDTO.getRepId());
         calculateContributionDTO.setRepOrderDTO(repOrderDTO);
 
         if (CaseType.APPEAL_CC.equals(calculateContributionDTO.getCaseType())) {
-            response = appealContributionService.calculateAppealContribution(calculateContributionDTO);
+            return appealContributionService.calculateAppealContribution(calculateContributionDTO);
         } else {
-            response = getCalculateContributionResponse(calculateContributionDTO, repOrderDTO);
+            return getCalculateContributionResponse(calculateContributionDTO, repOrderDTO);
```
[the remark this answers is not in the document]
Comment (venkat980) [reply crown-court-contribution/src/main/java/uk/gov/justice/laa/crime/contribution/service/MaatCalculateContributionService.java:153]: Done

File: crown-court-contribution/src/main/java/uk/gov/justice/laa/crime/contribution/builder/MaatCalculateContributionResponseMapper.java [hunk 3]
```diff
@@ -2,26 +2,50 @@
...
+                                                    ContributionResponseDTO contributionResponseDTO) {
+        ApiMaatCalculateContributionResponse response = new ApiMaatCalculateContributionResponse()
+                .withContributionCap(result.contributionCap())
+                .withEffectiveDate(DateUtil.convertDateToDateTime(result.effectiveDate()))
+                .withTotalMonths(result.totalMonths())
+                .withMonthlyContributions(result.monthlyAmount())
+                .withUpfrontContributions(result.upfrontAmount())
+                .withUpliftApplied(result.isUplift() ? "Y" : "N")
+                .withBasedOn(result.basedOn());
+        if (createdContribution != null) {
```
[the remark this answers is not in the document]
Comment (venkat980) [reply crown-court-contribution/src/main/java/uk/gov/justice/laa/crime/contribution/builder/MaatCalculateContributionResponseMapper.java:26]: Done

File: crown-court-contribution/src/main/java/uk/gov/justice/laa/crime/contribution/service/MaatCalculateContributionService.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (venkat980) [reply crown-court-contribution/src/main/java/uk/gov/justice/laa/crime/contribution/service/MaatCalculateContributionService.java]: Done

**Review by karlbaker02 — APPROVED** [review]
Looks good, thanks! 🥇

**Pull request merged**